RecoTracker/TkDetLayers/src/Phase2OTtiltedBarrelLayer.cc

```diff
@@ -0,0 +1,150 @@
+#include "Phase2OTtiltedBarrelLayer.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DetGroupMerger.h"
+#include "CompatibleDetToGroupAdder.h"
+
+#include "TrackingTools/DetLayers/interface/DetLayerException.h"
+#include "TrackingTools/DetLayers/interface/MeasurementEstimator.h"
+#include "TrackingTools/DetLayers/interface/CylinderBuilderFromDet.h"
+#include "Phase2OTEndcapLayerBuilder.h"
+
+//#include "DataFormats/SiPixelDetId/interface/PixelBarrelNameUpgrade.h"
+
+using namespace std;
+
+typedef GeometricSearchDet::DetWithState DetWithState;
+
+Phase2OTtiltedBarrelLayer::Phase2OTtiltedBarrelLayer(std::vector<const Phase2OTBarrelRod*>& innerRods,
+			                             std::vector<const Phase2OTBarrelRod*>& outerRods,
+                                                     vector<const Phase2OTEndcapRing*>& negRings, 
+                                                     vector<const Phase2OTEndcapRing*>& posRings) : 
+  Phase2OTBarrelLayer(innerRods,outerRods),
+  theNegativeRingsComps(negRings.begin(),negRings.end()),
+  thePositiveRingsComps(posRings.begin(),posRings.end())
+{
+
+  std::vector<const GeometricSearchDet*> theComps;
+  theComps.assign(Phase2OTBarrelLayer::components().begin(),Phase2OTBarrelLayer::components().end());
+  theComps.insert(theComps.end(),negRings.begin(),negRings.end());
+  theComps.insert(theComps.end(),posRings.begin(),posRings.end());
+
+  for(vector<const GeometricSearchDet*>::const_iterator it=theComps.begin();
+      it!=theComps.end();it++){  
+    theBasicComps.insert(theBasicComps.end(),	
+			 (**it).basicComponents().begin(),
+			 (**it).basicComponents().end());
+  }
+
+  BarrelDetLayer::initialize();
+  theCylinder = cylinder( theComps );
+
+  //--------- DEBUG INFO --------------
```

Comment:
since LogTrace and LogDebug are compiled only if EDM_ML_DEBUG is defined,
it makes sense to put all code that does only LogDebug or LogTrace inside some loops altogether inside ``` #ifdef EDM_ML_DEBUG ```